### ISSUE TITLE
Forgot Password Component

### ISSUE THREAD

**Issue by rdotjain** [Servatom/notefy#129]
Server endpoint to request for new password email is `/api/users/forgot_password/`
and to set new password is `/api/users/new_password/`

Create frontend component for the same.

**Comment by tchestnut85** [comment]
Hi @rdotjain , I'm still working on this but I'm a little confused and have some questions about the endpoint `/api/users/forgot_password/`. 
Is this to be used for a user entering their email in an input then they would be sent an email with link to the forgot password component to set their password?
I have a component to set a new password using the `/api/users/new_password/` endpoint done, just not sure about how to implement the other endpoint.

**Comment by tchestnut85** [comment]
@rdotjain Got it, thanks!

**Comment by tchestnut85** [comment]
Hi @rdotjain , I opened a PR for this but I think for checking if the email verification was successful I need some suggestion on how to check for that correctly. If you're able to take a look when you have time and give some guidance, that would be awesome.
Thank you!

**Comment by rdotjain** [comment]
Reviewed your PR @tchestnut85 !
There's already a check on the backend for email verification, I suppose you want this data to be available at the frontend as well? If that's the case, we can pass an `email_verified` field in the response at `/api/users/forgot_password/` endpoint.

**Comment by tchestnut85** [comment]
@rdotjain Yes that's what I was thinking, to be able to check that it's verified. Thank you!

**Comment by tchestnut85** [comment]
So I keep getting 404 errors when making the request to `/api/users/forgot_password/`. 
I tried `fetch(URL + '/api/users/forgot_password/' + email)` and also tried `fetch(URL + '/api/users/forgot_password/', {method: 'POST', body: email })` to send the submitted email in the body, but still no luck. Any tips on how I should make the request?
Sorry for the hassle ☹️

**Comment by yash22arora** [comment]
Hey @tchestnut85 , You could try something like this: 

 
``` js

const forgetPasswordHandler=(email)=>
{
    var myHeaders = new Headers();
    myHeaders.append("Content-Type", "application/json");
    
    var raw = JSON.stringify({
          "email": email
    });
    
    var requestOptions = {
          method: 'POST',
          headers: myHeaders,
          body: raw,
          redirect: 'follow'
    };
    
    fetch(`${URL}/api/users/forgot_password/`, requestOptions)
      .then(response => response.json())
      .then(result => console.log(result))
      .catch(error => console.log('error', error));

}
```

This returns the following when a valid email is enetered: 
```json
{
    "detail": "Email sent"
}
```

And this when an invalid email is enetred: 
```json
{
    "error": "Email not found"
}
```


Let us know if you still face an issues

**Comment by yash22arora** [comment]
> So I keep getting 404 errors when making the request to `/api/users/forgot_password/`. I tried `fetch(URL + '/api/users/forgot_password/' + email)` and also tried `fetch(URL + '/api/users/forgot_password/', {method: 'POST', body: email })` to send the submitted email in the body, but still no luck. Any tips on how I should make the request? Sorry for the hassle ☹️
> 
> Edit: When using 'POST' I'm getting this message from the response, although I'm sending JSON. `detail: "Unsupported media type \"text/plain;charset=UTF-8\" in request."`
> 
> ```js
> const submitEmail = email => fetch(`${URL}${FORGOT_PW_ENDPOINT}`, {
>   method: 'POST',
>   body: JSON.stringify(email),
>   'Content-Type': 'application/json',
> })	
> ```

I think you not passing the key-value pair in the body and directly passing email as value to the body attribute is making the difference. The rest of the code is almost the same. Also, the `Content-type` shall be passed in the header. Try the above code snippet and let us know if it worked ;)

**Comment by tchestnut85** [comment]
Thanks @yash22arora ! Adding the headers object and fixing the email key/value pair made it work.

**Comment by tchestnut85** [comment]
@yash22arora Sure thing, opened PR #141 to be merged to the dev branch. If anything need to be fixed or changed please let me know. Thank you!
